bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/api/JuiceNetApi.java

```diff
@@ -0,0 +1,230 @@
+        if (response.getStatus() != HttpStatus.OK_200) {
+            throw new JuiceNetApiException("Unable to retrieve device list, please check configuration (HTTP code: "
+                    + response.getStatus() + ").");
+        }
+
+        logger.trace("{}", response.getContentAsString());
+
+        JsonObject jsonResponse = JsonParser.parseString(response.getContentAsString()).getAsJsonObject();
+        boolean success = jsonResponse.get("success").getAsBoolean();
+
+        if (!success) {
+            throw new JuiceNetApiException("getDevices from JuiceNet failed, please check configuration.");
+        }
+
+        final JuiceNetApiDevice[] listDevices = new Gson().fromJson(jsonResponse.get("units").getAsJsonArray(),
+                JuiceNetApiDevice[].class);
+
+        return Arrays.asList(listDevices);
+    }
+
+    public JuiceNetApiDeviceStatus queryDeviceStatus(String token)
+            throws JuiceNetApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        Map<String, Object> params = new HashMap<>();
+        ContentResponse response;
+
+        params.put("cmd", "get_state");
+        params.put("account_token", apiToken);
+        params.put("device_id", bridgeUID.getAsString());
+        params.put("token", token);
+
+        response = httpApi.httpPost(API_DEVICE, params);
+
+        if (response.getStatus() != HttpStatus.OK_200) {
+            throw new JuiceNetApiException("Unable to retrieve device status, please check configuration (HTTP code: "
+                    + response.getStatus() + ").");
+        }
+
+        logger.trace("{}", response.getContentAsString());
+
+        JsonObject jsonResponse = JsonParser.parseString(response.getContentAsString()).getAsJsonObject();
+        boolean success = jsonResponse.get("success").getAsBoolean();
+
+        if (!success) {
+            throw new JuiceNetApiException("getDeviceStatus from JuiceNet failed, please check configuration.");
+        }
+
+        final JuiceNetApiDeviceStatus deviceStatus = new Gson().fromJson(jsonResponse, JuiceNetApiDeviceStatus.class);
+
+        return Objects.requireNonNull(deviceStatus);
+    }
+
+    public JuiceNetApiInfo queryInfo(String token)
+            throws IOException, InterruptedException, JuiceNetApiException, TimeoutException, ExecutionException {
+        Map<String, Object> params = new HashMap<>();
+        ContentResponse response;
+
+        params.put("cmd", "get_info");
+        params.put("account_token", apiToken);
+        params.put("device_id", bridgeUID.getAsString());
+        params.put("token", token);
+
+        response = httpApi.httpPost(API_DEVICE, params);
+
+        if (response.getStatus() != HttpStatus.OK_200) {
+            throw new JuiceNetApiException("Unable to retrieve device info, please check configuration (HTTP code: "
+                    + response.getStatus() + ").");
+        }
+
+        logger.trace("{}", response.getContentAsString());
```

Comment:
Some overhead could be avoided by storing content in local String as `getContentAsString()` as called again in next line.